test/integration/http/websockets-connect-test.js

```diff
@@ -0,0 +1,121 @@
+let { join } = require('path')
+let test = require('tape')
+let http = require('http')
+let Websocket = require('ws')
+let sut = join(process.cwd(), 'src')
+let sandbox = require(sut)
+
+let mock = join(process.cwd(), 'test', 'mock')
+let sandboxPort = Number(process.env.PORT || 3333)
+let url = `ws://localhost:${sandboxPort}`
+
+async function* sideChannel (port = 3433) {
+  let events = []
+  console.log('starting server')
+  let httpServer = http.createServer((req, res) => {
+    console.log('on request')
+    let inputData = []
+    req.on('data', data => {
+      inputData.push(data)
+    })
+    req.on('end', () => {
+      res.writeHead(200)
+      console.log('on req end')
+      res.end(() => {
+        console.log('on resp end')
+        const postData = Buffer.concat(inputData).toString()
+        events.push(postData)
+      })
+    })
+  })
+
+  await new Promise((resolve, reject) => httpServer.listen(port, (err) => err ? reject(err) : resolve()))
+  console.log('server started')
+  yield 'start'
+  while (true) {
+    if (events.length > 0) {
+      try {
+        yield* events
+      }
+      catch (e) {
+        break
+      }
+      events = []
+    }
+    else {
+      console.log('waiting for events in sideChannel')
+      await new Promise(resolve => httpServer.once('request', (req, res) => res.once('close', resolve)))
+      console.log('got event!')
+    }
+  }
+  httpServer.close()
+}
+
+let expectFunctionBasics = (t, expectedFunctionName) => websocketMessage => {
```

Comment:
They're accomplishing the same thing, so either way we should def combine! Per above, I'm not a huge fan of the generator syntax, but the sidechannel server works. Any reason not to just continue using the `arc.ws.send` approach for data egress instead of kicking up a new webserver? Nothing to say the `$disconnect` event can't also do what we're doing here: https://github.com/architect/sandbox/blob/master/test/mock/normal/src/ws/custom-path/index.js#L4-L10